api/src/main/java/gov/cms/ab2d/api/security/OktaTokenGeneratorImpl.java

```diff
@@ -0,0 +1,48 @@
+package gov.cms.ab2d.api.security;
+
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Base64;
+
+@Service
+@Slf4j
+public class OktaTokenGeneratorImpl implements OktaTokenGenerator {
+
+    private final HttpClient httpClient = HttpClient.newBuilder()
+            .version(HttpClient.Version.HTTP_2)
+            .build();
+
+    public String generateToken(String clientID, String clientSecret) throws URISyntaxException, IOException, InterruptedException, JSONException {
+        log.info("Received request to generate token");
+
+        String authString = Base64.getEncoder().encodeToString((clientID + ":" + clientSecret).getBytes());
+
+        HttpRequest request = HttpRequest.newBuilder()
+                .uri(new URI("https://test.idp.idm.cms.gov/oauth2/aus2r7y3gdaFMKBol297/v1/token?grant_type=client_credentials&scope=clientCreds"))
```

Comment:
Okta URL should come from the application configuration, `api.okta-jwt-issuer` specifically.